incubator/hnc/internal/validators/object.go

```diff
@@ -111,18 +112,31 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	// If the object wasn't and isn't inherited, we will check to see if the
 	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
+		// check selector format
+		oldSelectorStr, _ := selector.GetSelectorAnnotation(oldInst)
+		newSelectorStr, _ := selector.GetSelectorAnnotation(inst)
+		sel := selector.GetSelectorFromString(log, newSelectorStr)
+		if oldSelectorStr != newSelectorStr && sel == nil {
```

Comment:
Maybe comment on _why_ we compare the old and new ones first.

Reply:
done